### PR TITLE
Build and archive software update image

### PR DESCRIPTION
**Pull request by sergeevaeka**
Add archiving .swu for ARP, intel, RPI. Add building update image for ARP.

### CODE REVIEW

**Comment by TSAnsari** [comment]
Improve commit message. Its too long. Try to keep the first line around 50 characters. Details can be provided in the next paragraph. For example,

"Archive software update .swu image

Archives .swu image for Intel, rpi and arp for nightly builds"

File: ci-scripts/yocto.groovy [hunk 1]
```diff
@@ -255,7 +256,8 @@ void buildManifest(String variantName, String imageName, String layerToReplace="
 
         // Build the images
         try {
-            boolean buildUpdate = variantName.startsWith("rpi")
+            boolean buildUpdate = variantName.startsWith("rpi") || variantName.startsWith("intel") || variantName.startsWith("arp")
+            echo "testing for arp building"
```
Comment (TSAnsari) [inline ci-scripts/yocto.groovy]: remove this line. This seemed to be added for debugging purpose.

File: ci-scripts/yocto.groovy [hunk 2]
```diff
@@ -267,6 +269,9 @@ void buildManifest(String variantName, String imageName, String layerToReplace="
                 }
             }
 
+        }  catch(e) {
+            echo "Building is failing!!!!"
```
Comment (TSAnsari) [inline ci-scripts/yocto.groovy]: "Bitbake process failed" or something similar

File: ci-scripts/yocto.groovy [hunk 3]
```diff
@@ -163,6 +163,7 @@ void archiveImagesAndSDK(String yoctoDir, String suffix) {
 
         sh "rm -rf ${artifactDir}"
         sh "mkdir ${artifactDir}"
+        sh "echo ${artifactDir}"
```
Comment (TSAnsari) [inline ci-scripts/yocto.groovy]: remove. probably added for debugging.

File: ci-scripts/copy_to_archive [hunk 4]
```diff
@@ -40,18 +40,24 @@ fi
 if [ -d ${IMAGE_DIR}/${INTEL_ARCH} ]; then
     echo "Archiving Intel image to $ARCHIVE_DIR"
     mv ${IMAGE_DIR}/${INTEL_ARCH}/${IMAGE_BASENAME}-*rootfs.wic $ARCHIVE_DIR
+    mv ${IMAGE_DIR}/${INTEL_ARCH}/${IMAGE_BASENAME}-*20*.swu $ARCHIVE_DIR
```
Comment (TSAnsari) [inline ci-scripts/copy_to_archive]: Why is there a need for "*20*". 

only "-*.swu" would do the trick.

File: ci-scripts/copy_to_archive [hunk 5]
```diff
@@ -40,18 +40,24 @@ fi
...
 fi
 
 # If there is an ARP directory
+
+echo "${IMAGE_DIR}/${ARP_ARCH}"
```
Comment (TSAnsari) [inline ci-scripts/copy_to_archive]: remove.

File: ci-scripts/copy_to_archive [hunk 6]
```diff
@@ -40,18 +40,24 @@ fi
...
+
 if [ -d ${IMAGE_DIR}/${ARP_ARCH} ]; then
     echo "Archiving ARP image to $ARCHIVE_DIR"
     mv ${IMAGE_DIR}/${ARP_ARCH}/${IMAGE_BASENAME}-*rootfs.wic $ARCHIVE_DIR
+    mv ${IMAGE_DIR}/${ARP_ARCH}/${IMAGE_BASENAME}-*20*.swu $ARCHIVE_DIR
```
Comment (TSAnsari) [inline ci-scripts/copy_to_archive]: same as above

File: ci-scripts/copy_to_archive [hunk 7]
```diff
@@ -40,18 +40,24 @@ fi
...
 fi
 
 # If there is an RPI directory
 if [ -d ${IMAGE_DIR}/${RPI_ARCH} ]; then
     echo "Archiving RPI image to $ARCHIVE_DIR"
     mv ${IMAGE_DIR}/${RPI_ARCH}/${IMAGE_BASENAME}-*rootfs.rpi-sdimg $ARCHIVE_DIR
+    mv ${IMAGE_DIR}/${RPI_ARCH}/${IMAGE_BASENAME}-*20*.swu $ARCHIVE_DIR
```
Comment (TSAnsari) [inline ci-scripts/copy_to_archive]: same as above

File: ci-scripts/copy_to_archive (hunk 4 above)
Comment (TSAnsari) [reply ci-scripts/copy_to_archive]: but the "-" before 20 is enough to distinguish b/w symlink and the timestamped image. The symlink doesn't have "-".

File: ci-scripts/yocto.groovy (hunk 1 above)
Comment (sergeevaeka) [reply ci-scripts/yocto.groovy]: Done

File: ci-scripts/yocto.groovy (hunk 2 above)
Comment (sergeevaeka) [reply ci-scripts/yocto.groovy]: Done

File: ci-scripts/yocto.groovy (hunk 3 above)
Comment (sergeevaeka) [reply ci-scripts/yocto.groovy]: Done

File: ci-scripts/copy_to_archive (hunk 5 above)
Comment (sergeevaeka) [reply ci-scripts/copy_to_archive]: Done

File: ci-scripts/copy_to_archive (hunk 4 above)
Comment (TSAnsari) [reply ci-scripts/copy_to_archive]: Please fix this one too

Comment (sergeevaeka) [reply ci-scripts/copy_to_archive]: In practice it is not working, I tried but it is still archiving both.

**Comment by sashko** [comment]
so... this does not have SOB, author's name or correct email and yet... it was merged :/

**Pull request merged**